feature_details.md

```diff
@@ -0,0 +1,174 @@
+
+* The entry point for this API is document.documentTransition. The previous Document provides a url keyed list of elements, referred to as shared elements, which could be animated during a transition. There are a couple of important assumptions here :
+
+    * Root transitions are created by using the `html` element in this API. There may be aspects of the transition where the root element will have special behaviour. But the proposal tries to align it closely with nested elements.
+
+    * The type of navigations the previous Document will be allowed to define a transition for should be similar to the ones supported by app-history’s [navigate event](https://github.com/WICG/app-history#restrictions-on-firing-canceling-and-responding). The most likely exception will be the back/forward buttons. So a common pattern would be to use this API in response to the navigate event.
+
+* When a navigation is initiated to this url, the UA caches a static pixel snapshot for the DOM sub-tree of each shared element. If there is any hierarchical relationship between these elements, they are captured as separate snapshots to allow them to animate independently. The parent element's snapshot excludes content for any embedded shared elements, drawing the background which was earlier occluded by the child element.
+
+* The feature supports a subset of [animatable properties](https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_animated_properties) that can be interpolated during a transition. For each of those properties, the UA caches their start value to animate them during the transition. In addition to this metadata, we allow the developer to provide opaque contextual information (a unique identifier for the element, navigation trigger, scroll offset, site version etc.), referred to as propertyMap, about these elements to configure the transition.
+
+```
+document.documentTransition.setSameOriginTransitions(
+  {“foo.com/a.html” : [{element : document.html, propertyMap : {“id” : “root”}},
+                                 {element : document.getElementById(‘shared’), propertyMap : {“id” : “shared”}],
+   “foo.com/b.html” : [{element : document.html}]
+  });
+```
+
+* When the new Document loads, the developer can register an `EventListener` to be notified if a transition was initiated by the previous Document. The event, referred to as `handleTransition`, provides the metadata for elements captured from the previous Document and the previous URL, a detailed explanation is at [Referencing Elements](#referencing-elements). The aim of the `handleTransition` event is to allow the developer to define the animations for a transition using information from both Documents. For instance, a paired element transition may be dropped if the corresponding element in the new Document renders offscreen.
```

Comment:
> new Document loads

Do you mean the onload event fires? I think that may be too late to start a transition.

Reply:
This is a detail that still needs to be clarified. The exact stage in the navigation's lifecycle when this event is dispatched.